blockchain-albatross/src/blockchain.rs

```diff
@@ -592,6 +599,72 @@ impl Blockchain {
                 return Err(PushError::InvalidBlock(BlockError::InvalidJustification));
             }
 
+            // Check if there are two blocks in the same slot. Since we already
+            // verified the validator for the current slot, this is enough to
+            // check for fork proofs. A fork proof doesn't apply to the genesis
+            // block, so skip block 0.
+            if block.block_number() != 0 {
+                // Count the microblocks after the last macroblock
+                let count = block.block_number() - policy::macro_block_before(block.block_number());
+                let microblocks: Vec<Block> = self.chain_store.get_blocks_backward(block.parent_hash(), count - 1, false, Some(&read_txn));
+
+                // Get the microheader from the block
+                let header = block.header();
+                let microheader1;
+
+                if let BlockHeader::Micro(microheader) = header {
+                    microheader1 = microheader;
+                } else {
+                    unreachable!();
+                };
+
+                // Get the justification for the block. We asume that the
+                // validator's signature is valid.
+                let justification1;
+                if let Block::Micro(ref micro_block) = block {
+                    justification1 = micro_block.justification.signature;
+                } else {
+                    unreachable!();
+                };
+
+                for microblock in microblocks.iter() {
```

Comment:
If we fetch all micro blocks at the given height, we still need the iteration here. :+1: